src/ghc.jl

```diff
@@ -0,0 +1,133 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    GHC(kernel,merge_crit;k=35,vars=nothing)
+
+A method for partitioning spatial data into `k` clusters 
+using Geostatistical Hierarchical Clustering (GHC). This method generates
+clusters by assigning a unique cluster to each datum, 
+iteratively joining the two clusters which are most similar
+according to a kernel estimator on the data's spatial 
+dependence, and finally stopping this process at the step 
+where the clusters' silhouette index is optimized.
+
+## Parameters
+
+* `kernel`      - type of kernel used for estimator
+* `merge_crit`  - strategy to specify difference between clusters
+* `k`           - minimum number of observations contained in support 
+                    of kernel function centered at each point. 
+                    Defaults to 35.
+* `vars`        - Variables (or features) to consider (default to all)
```

Comment:
We should basically provide a `kernel` option as either a `Symbol` or a lambda function. Internally we always store a function as the field parameter but the constructor could check if the user passed a symbol and would then convert to a specific function. The other parameters should basically be the same parameters of the inner hclust call.